data_selector.py

```diff
@@ -122,40 +125,78 @@ def __load_feature(self, feature):
-                    if cfg.get_filter_value():
-                        filter_expr += " and {}='{}'".format(cfg["feature_filter_column"],
-                                                             cfg.get_filter_value())
-
-                        title = cfg.get_filter_value()
-                    else:
-                        title = cfg["name"]
-
-                    f = None
-                    for f in data_l.getFeatures(req):
-                        pass
-                    if f is None:
-                        return
-                    uom = cfg.get_uom()
-                    data = LayerData(data_l, cfg["event_column"], cfg["value_column"],
-                                     filter_expression=filter_expr, uom=uom)
+                    data = LayerData(
+                        data_l,
+                        cfg["event_column"],
+                        cfg["value_column"],
+                        filter_expression=filter_expr,
+                        uom=uom
+                    )
                     uom = data.uom()
-
+                    self.__viewer.add_data_cell(
+                        data,
+                        title,
+                        uom,
+                        station_name=feature_name,
+                        config=cfg
+                    )
+                    self.__viewer.add_scale()
                 if cfg["type"] == "continuous":
-                    uom = cfg["uom"]
                     fids = [f.id() for f in data_l.getFeatures(req)]
-                    data = FeatureData(data_l, cfg["values_column"], feature_ids=fids,
-                                       x_start_fieldname=cfg["start_measure_column"],
-                                       x_delta_fieldname=cfg["interval_column"])
-
-                if hasattr(self.__viewer, "add_data_column"):
-                    self.__viewer.add_data_column(data, title, uom,
-                                                  station_name=feature_name,
-                                                  config=cfg)
-                if hasattr(self.__viewer, "add_data_row"):
-                    self.__viewer.add_data_row(data, title, uom,
-                                               station_name=feature_name,
-                                               config=cfg)
+                    data = FeatureData(
+                        data_l,
+                        cfg["values_column"],
+                        feature_ids=fids,
+                        x_start_fieldname=cfg["start_measure_column"],
+                        x_delta_fieldname=cfg["interval_column"]
+                    )
+                    self.__viewer.add_data_cell(
+                        data,
+                        title,
+                        uom,
+                        station_name=feature_name,
+                        config=cfg
+                    )
+                elif cfg["type"] == "cumulative":
+                    data = IntervalData(
```

Comment:
Sorry to go back to this merged feature... I'm looking at the source code to understand how it does work for me.

Not sure to understand this part of the code: here, is the `data` variable used in any way?

Reply:
Thanks for the review. IntervalData is actually not used anymore

Reply:
Fixed in https://github.com/Oslandia/QGeoloGIS/commit/37cee35695dddc9fad15d81bbcfc544680435222